components/kyma-environment-broker/internal/kubeconfig/handler.go

```diff
@@ -0,0 +1,131 @@
+package kubeconfig
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/httputil"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+	"github.com/pivotal-cf/brokerapi/v8/domain"
+	"github.com/sirupsen/logrus"
+)
+
+const attachmentName = "kubeconfig.yaml"
+
+//go:generate mockery -name=KcBuilder -output=automock -outpkg=automock -case=underscore
+
+type KcBuilder interface {
+	Build(*internal.Instance) (string, error)
+}
+
+type Handler struct {
+	kubeconfigBuilder KcBuilder
+	instanceStorage   storage.Instances
+	operationStorage  storage.Operations
+	log               logrus.FieldLogger
+}
+
+func NewHandler(storage storage.BrokerStorage, b KcBuilder, log logrus.FieldLogger) *Handler {
+	return &Handler{
+		instanceStorage:   storage.Instances(),
+		operationStorage:  storage.Operations(),
+		kubeconfigBuilder: b,
+		log:               log,
+	}
+}
+
+func (h *Handler) AttachRoutes(router *mux.Router) {
+	router.HandleFunc("/kubeconfig/{operation_id}/instance/{instance_id}", h.GetKubeconfig).Methods(http.MethodGet)
+}
+
+type InfoResponse struct {
+	Message string
+}
+
+type ErrorResponse struct {
+	Error string
+}
+
+func (h *Handler) GetKubeconfig(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	instanceID := vars["instance_id"]
+	operationID := vars["operation_id"]
+
+	if !h.isValidUUID(operationID) {
+		h.handleErrorResponse(w, http.StatusBadRequest, errors.New("wrong operationID"))
+		return
+	}
+
+	if !h.isValidUUID(instanceID) {
```

Comment:
KEB allows the instance ID to be not valid UUID, so why do we need to validate it here?

Reply:
right, I forgot we support not uuid type of ID